client/src/App.tsx

```diff
@@ -1382,16 +1391,21 @@ function App() {
 					isMovingBoardObject={!!movingBoardItem}
 				/>
 			)}
-
 			<Modal
-				onClose={() => setIsModalOpen(false)}
+				onClose={() => setModalState('none')}
 				className="modal-container"
-				open={isModalOpen}
+				open={modalState === 'new-room' || modalState === 'enter-room'}
 			>
-				<NewChatroom
-					onClickCancel={() => setIsModalOpen(false)}
+				<div >
```

Comment:
use <> instead of <div> 

https://reactjs.org/docs/fragments.html#short-syntax